### DISCUSSION TITLE
ISC-DCP and NEW Kea DHCP server

### DISCUSSION THREAD

**Discussion by hanno-froehlich** [ltsp/ltsp#900] [Ideas]
Thanks a lot for your great LTSP-work! I use it in school-environment. I have one Question to ISC-DHCP and Kea DHCP server. (https://www.isc.org/kea/)
ISC has announced the end of maintenance for ISC DHCP as of the end of 2022. Can I use the new kea-dhcp-server as it described at your LTSP-Page?
https://ltsp.org/guides/isc-dhcp-server/
Many greetings from germany.
hanno

**Comment by alkisg** [comment]
Hi, LTSP defaults to dnsmasq, so it's not impacted at all by isc-dhcp getting EOL'ed.
I've only spent a few minutes testing isc-dhcp in order to write that guide.
I haven't even seen kea yet so I don't have a guide handy. If you do write one, I can upload it.

**Reply by hanno-froehlich** [reply]
OK. I will test it. KEA looks like a little complicated. isc-dhcp (your script) works realy fine in my environment!
hanno

**Comment by hanno-froehlich** [comment]
Hi LTSP-Team.
ISC-DHCP for static-IP-Client-Adresses is discontinued. KEA is the new DHCP-Server!
https://www.isc.org/kea/
Just use it, if you need STATIC-IP-Adresses for your clients! E.g. if you want to use Veyon  Cross-platform computer control and classroom management (https://veyon.io/en/).
Use Epoptes (https://epoptes.org/documentation/ltsp-groups/) for an easy way to control clients over mac-adress!
I found a first solution for the ISC-DHCP update problem to KEA. Tested in VirtualBox with PFSEnse (2 NICS: Nat and intern network) and Linux Mint 21.3 (one nic: automatic dhcp!).
First you have to use PFSense for NAT and intern Network. Set the NAT  (WAN-Interface) and uncheck (block) private networks and bogon networks.
![grafik](https://github.com/user-attachments/assets/1f766fa6-a9ce-42a5-83f0-516e841f9bde)
Go to SYSTEM/PACKAGE Manager and install TFTP!
![grafik](https://github.com/user-attachments/assets/112ffd44-b102-44db-851d-a67d2ed860a1)
Go to SYSTEM/ADVANCED/NETWORKING and uncheck isc dhcp and check KEA.
![grafik](https://github.com/user-attachments/assets/57a791d8-6bb8-42ee-9ab8-a1210755af9f)
SAVE!
Go to SERVICES/DHCP SERVER/LAN. Check enable.
![grafik](https://github.com/user-attachments/assets/65086add-7355-4c30-93f7-3d535933237d)
Give the Address Pool a range you need...
![grafik](https://github.com/user-attachments/assets/c9f00f4e-eb92-45ee-99c3-1ff760d4a7bc)
Scroll down. Give the TFTP-Server a IP outside the POOL.
![grafik](https://github.com/user-attachments/assets/09fa10c2-442c-467f-a855-6a166f906349)
Enable Network Booting and set the Next Server (IP AND DEFAULT BIOS NAME: ltsp/undionly.kpxe).
![grafik](https://github.com/user-attachments/assets/c2451df6-3e80-479a-be63-cc884cd08305)
SAVE!
ADD Static Map for LTSP-Server. Down an Right.
![grafik](https://github.com/user-attachments/assets/402724ed-646b-48ae-8877-74509ebd96ac)
![grafik](https://github.com/user-attachments/assets/f7773b3c-0cec-45a3-9ba0-e8035dcecc24)
SAVE! APPLY CHANGES!
Restart your Linux-LTSP-Server to get the new StaticIP-Adress. Update your LTSP-Server and your LTSP-System.
Boot Client:
![grafik](https://github.com/user-attachments/assets/4d0c6dd6-7b91-4396-a6c1-4fc95dd94df5)
![grafik](https://github.com/user-attachments/assets/47e57bea-d6ba-4480-9a16-9d29e0705e40)

Good Luck!
hanno
